editor/gui/scene_tree_editor.cpp

```diff
@@ -1439,9 +1439,32 @@ void SceneTreeEditor::rename_node(Node *p_node, const String &p_name, TreeItem *
 		item = _find(tree->get_root(), p_node->get_path());
 	}
 	ERR_FAIL_NULL(item);
-	String new_name = p_name.validate_node_name();
+	bool check_for_unique_name_token = p_name[0] == '%';
```

Comment:
Are we sure that `rename_node` is never/can never be called with an empty `p_name`?

Reply:
Renaming to empty name does not seem to crash, but this check could be easily tweaked just to be safe.
```suggestion
	bool check_for_unique_name_token = !p_name.is_empty() && p_name[0] == '%';
```